### PR TITLE
HDDS-4508. Add total container count to SCMContainerMetrics

### PR DESCRIPTION
**Pull request by sky76093016**
## What changes were proposed in this pull request?

Add total container count to SCMContainerMetrics.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-4508

## How was this patch tested?

Tests passed.

### CODE REVIEW

**Review by xiaoyuyao — APPROVED** [review]
LGTM, +1.

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMContainerMetrics.java [hunk 1]
```diff
@@ -84,6 +84,14 @@ public void getMetrics(MetricsCollector collector, boolean all) {
...
         .addGauge(Interns.info("DeletedContainers",
             "Number of containers in deleted state"),
+            stateCount.get(DELETED.toString()))
+        .addGauge(Interns.info("TotalContainers",
+            "Number of all containers"),
+          stateCount.get(OPEN.toString())+
+            stateCount.get(CLOSING.toString())+
+            stateCount.get(QUASI_CLOSED.toString())+
+            stateCount.get(CLOSED.toString())+
+            stateCount.get(DELETING.toString())+
```
Comment (adoroszlai) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMContainerMetrics.java:94]: Nit: Can this be converted to a loop over all states summing individual values?  That way if a new state is introduced, no further change in counting totals will be required.

**Review by adoroszlai — CHANGES_REQUESTED** [review]
Thanks @sky76093016 for working on this.

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMContainerMetrics.java (hunk 1 above)
Comment (sky76093016) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMContainerMetrics.java]: Thanks @adoroszlai  for the suggestion,I tried to modify it.

**Review by adoroszlai — APPROVED** [review]
Thanks @sky76093016 for updating the patch.

**Comment by adoroszlai** [comment]
Thanks @sky76093016 for the contribution, @xiaoyuyao for the review.

**Pull request merged**